### PR TITLE
Added4012Q

### PR DESCRIPTION
**Pull request by PhilDeakin**
Added4012Q

### CODE REVIEW

File: cin_validator/rules/cin2022_23/rule_4012Q.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    Within a <CINPlanDates> group, <CINPlanStartDate> (N00689) should not be the same as the <CINPlanEndDate> (N00690)
+    """
+    
+    df = df[df['CINPlanStartDate'] == df['CINPlanEndDate']]
+
+    failing_indices = df.index
+
+    rule_context.push_issue(
+        table=CINplanDates, field=CINPlanStartDate, row=failing_indices
+    )
```
Comment (tab1tha) [inline cin_validator/rules/cin2022_23/rule_4012Q.py:33]: Ideally, when multiple columns are compared, we want to be able to push multiple columns in a way that shows the relationship between them. 
I'll put this on hold while we finalize plans on how to do that.

**Comment by tab1tha** [comment]
Hi @PhilDeakin ,
Since you attempted a type1 rule already, you can have an early sneak peek at the proposed structure for rules like these: https://github.com/data-to-insight/CIN-validator/blob/main/cin_validator/rules/cin2022_23/rule_8925.py

When a rule implementation interacts with multiple columns in a single table, we are choosing to call that a type1 rule.
The comments can guide you through implementing this pull request. More learning resources will come through next week.

See you !

**Comment by PhilDeakin** [comment]
@tab1tha I'll have a go later this week, sorry I've been given a piece of work to do that's taking up most of time at the moment so I haven't been able to do as much Python coding as usual & I can't make the meeting this afternoon. But.. I'm still on it.

**Comment by tab1tha** [comment]
Hi @PhilDeakin , 

That's totally okay !.

**Comment by PhilDeakin** [comment]
Merged on a different branch.

**Pull request closed without merging**